### PR TITLE
Add cache version to persisted state and ignore older versions

### LINKED ISSUE #93: Cache should be versioned

### PR DESCRIPTION
**Pull request by Bettelstab**
Resolves #93

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -18,35 +20,35 @@ export class Store {
...
+                // support legacy version of persisted contacts
+                if (persistedState.contacts) {
+                    initialState.contacts = Object.assign({}, persistedState.contacts);
+                }
             }
-        );
+        }
+
+        if (persistedContacts) {
+            initialState.contacts = Object.assign({}, persistedContacts);
```
Comment (sisou) [inline src/store.js]: There might be a reason why the code before was handling the contacts as inheriting from each other
```
Object.assign({}, persistedState.contacts || {}, persistedContacts || {})
```
I can't think of a reason now, but there was propably an edge case, maybe?

Would it be possible to do this inheritance here, too?

File: src/store.js [hunk 2]
```diff
@@ -71,6 +73,7 @@ export class Store {
         );
 
         const persistentState = {
+            cacheVersion: CACHE_VERSION,
```
Comment (sisou) [inline src/store.js]: Any arguments against calling it just `version` instead of cacheVersion?

Comment (Bettelstab) [reply src/store.js]: No arguments. Let's do it.

File: src/store.js (hunk 1 above)
Comment (Bettelstab) [reply src/store.js]: Don't know why you did it like that, but I'm pretty sure it's not necessary. Telling from https://github.com/nimiq/safe/commit/de849a882c0208fa3c7523b5771a3d5ce8f84787 there was never the possibility of both variants coexisting.

**Pull request merged**